content/2022/prepoznavanje-govora.md

```diff
@@ -2,3 +2,231 @@
+- Spektrogram se prvo obrađuje konvolucijom i ReLU-om
+- Smanjujemo veličinu obrađene slike pooling slojem
+- Ponavljamo ovaj proces
+
+Konvolucija (po čemu nastaje termin konvolucione neuronske mreže) u obradi slike je operator koji predstavlja sužavanje početne slike množenjem iste određenim filterom. 
+
+Konvolucija kao bitne detalje posmatra one koji su mnogo puta uhvaćeni u kernelu. Problem može da se desi kada kernel ne zahvata ivice dosta puta, te može mnogo da smanji određenu sliku, a samim tim i da se reši ivičnih detalja. Ako do te pojave dođe, koristi se tehnika koja se zove padding. 
+
+Padding označava dodavanje piksela na ivice. Samim tim, kada konvolucija radi svoj posao, ona će svojim kernelom mnogo puta pokriti tu površinu. 
+
+ReLU (rectified linear activation function / rectified linear unit) je funkcija koja negativnim vrednostima daje nulu, a pozitivne ostavlja kakve jesu. Time dobijamo nelinearan model.
+
+![Funkcija](static\images\fja.png)
+
+Kroz neuronsku mrežu se propušta već napravljen spektrogram, kao i labele tih spektrograma koje mreža treba da prepozna.
+
+Za treniranje mreže koriste se dve metode simultano: loss funkcija i back propagation.
+
+Cost funkcija (funkcija troškova ili gubitka) jeste funkcionalna veza željenog outputa i dobijenog outputa u funkciji, a loss funkcija je srednja vrednost svih cost funkcija.
+
+Najkorišćenija loss funkcija je Cross Entropy Loss. 
+Cross Entropy Loss radi tako što pokušava da minimizuje razliku između tačnih rezultata i verovatnoće predviđanja, to jest output.
+
+Formula po kojoj se računa Cross Entropy Loss je sledeća:
+
+$H_p(q)=-\frac{1}{N} \sum_{i=1}^N y_i \cdot \log \left(p\left(y_i\right)\right)+\left(1-y_i\right) \cdot \log \left(1-p\left(y_i\right)\right)$
+
+Back propagacija je metod smanjenja grešaka u CNN posmatranjem neophodnih promena u prethodnom sloju od aktivacije da bi se u određenom sloju neuroni aktivirali na određen način.
+
+Backpropagation prolazi krroz sve primere i traži sumu svih težina veza među neuronima.
+
+Težine se menjaju u cilju računanja dovoljno dobrog gradient descenta za traženje lokalnog / maksimalnog minimuma ove funkcije, to jest tačnu reč.
+
+![SGD](static\images\sgd.png)
+
+### Istraživanje i rezultati
+
+Rezultati su krajnje očekivani uzimajući u obzir veličinu baze koja je obrađivanja. Bez interaktivnog interfejsa, dosadašnji rezultati svode se na tačnost (accuracy) svake metode u radu. 
+
+![Rezultati](static\images\4.png)
+
+Iz tabele iznad može se uočiti kako rezultati dosta variraju jedni od drugih. Konvoluciona neuronska mreža daje maksimalnu preciznost u istim uslovima, dok SVM sa polinomijalnim kernelom daje minimalne. 
+
+Svoj potencijal SVM može da pokaže kada je lako odrediti kojoj labeli koji podatak pripada. U ovom slučaju, određene reči mogu lako da se pomešaju na spektrogramu, pa su neke vrednosti vrlo blizu odlučnoj granici i da pomute labele. Iz tog razloga, rezultati su veoma dobri za ovu metodu. 
```

Comment:
Zar ne rade svi klasifikatori najbolje kad je lako odrediti labelu podatka hahahahah?
Btw ne bih se upustao u analizu zasto je linearni kernel dao bolje rezultate od polinomskog. U teoriji to stvarno ne bi smelo da se desi, tkd je vrv neka glupa implementacijska greska